imap/index.c

```diff
@@ -6056,7 +6066,8 @@ EXPORTED int index_getsearchtext(message_t *msg, const strarray_t *partids,
     if (r == IMAP_OK_COMPLETED) r = 0;
 
     /* Log erroneous or partially indexed message */
-    if (r || (str.indexlevel & SEARCH_INDEXLEVEL_PARTIAL)) {
+    if ((r || (str.indexlevel & SEARCH_INDEXLEVEL_PARTIAL)) &&
+        !(flags & INDEX_GETSEARCHTEXT_NOLOG_PARTIALS)) {
```

Comment:
This looks like turning off logging of partials will also turn off logging of errors.  Is that intended?